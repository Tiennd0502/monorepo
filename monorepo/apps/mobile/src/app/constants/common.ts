export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
