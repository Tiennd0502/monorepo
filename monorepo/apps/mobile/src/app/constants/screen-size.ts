import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
