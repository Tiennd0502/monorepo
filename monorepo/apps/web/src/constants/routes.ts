export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
