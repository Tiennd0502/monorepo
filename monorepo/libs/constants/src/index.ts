export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
