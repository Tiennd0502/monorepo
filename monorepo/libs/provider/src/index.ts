export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
