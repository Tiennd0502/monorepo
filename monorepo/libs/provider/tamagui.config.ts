import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
