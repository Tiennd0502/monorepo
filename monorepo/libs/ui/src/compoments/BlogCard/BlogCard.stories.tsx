import { Meta, StoryObj } from '@storybook/react';
import { Stack } from 'tamagui';

import { BLOGS } from '@monorepo/mocks';

import BlogCard from '.';

const meta: Meta<typeof BlogCard> = {
  title: 'Components/BlogCard',
  component: BlogCard,
  decorators: [
    (Story) => (
      <Stack width={450} paddingTop={20}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof BlogCard>;

export const Default: Story = {
  args: {
    item: BLOGS[0],
    onPress: () => null,
  },
};

export const Favorite: Story = {
  args: {
    isReview: true,
    item: BLOGS[0],
    onPress: () => null,
  },
};
