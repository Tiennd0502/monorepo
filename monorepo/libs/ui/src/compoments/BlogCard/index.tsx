import { memo } from 'react';
import { Stack, XStack, Image } from 'tamagui';
import isEqual from 'react-fast-compare';

// Types
import { Blog } from '@monorepo/types';

// Constants
import { AVATAR_DEFAULT } from '@monorepo/constants';

// Components
import Text from '../Text';
import Rating from '../Rating';
import { shadows } from '../../themes';

interface BlogCardProps {
  isReview?: boolean;
  item: Blog;
  onPress: () => void;
}

const BlogCard = ({
  isReview = false,
  item: { price, title, image, evaluate, createdAt, description },
  onPress,
}: BlogCardProps) => (
  <Stack
    backgroundColor="$secondary"
    borderRadius="$2"
    padding="$5"
    style={shadows.card}
    onPress={onPress}
  >
    {isReview ? (
      <Image
        position="absolute"
        alignSelf="center"
        top={-20}
        width="$10"
        height="$10"
        borderRadius="$full"
        source={{ uri: image || AVATAR_DEFAULT }}
      />
    ) : (
      <XStack gap="$5" marginBottom="$5">
        <Image
          width="$17.5"
          height="$17.5"
          borderRadius="$2"
          source={{ uri: image || AVATAR_DEFAULT }}
        />
        <Stack flex={1} gap="$1.5">
          <Text
            size="extraMedium"
            color="$textQuaternary"
            numberOfLines={2}
            ellipsizeMode="tail"
          >
            {title}
          </Text>
          {price && (
            <Text color="$primary" fontWeight="bold">
              {price}
            </Text>
          )}
        </Stack>
      </XStack>
    )}
    <XStack justifyContent="space-between" marginBottom="$2">
      <Stack rowGap={5}>
        {isReview && (
          <Text color="$primary" numberOfLines={2} ellipsizeMode="tail">
            {title}
          </Text>
        )}
        <Rating value={evaluate} />
      </Stack>
      <Text fontSize="$3" color="$textLabel">
        {createdAt}
      </Text>
    </XStack>
    <Text
      color="$primary"
      numberOfLines={5}
      ellipsizeMode="tail"
      textAlign="justify"
    >
      {description}
    </Text>
  </Stack>
);

export default memo(BlogCard, isEqual);
