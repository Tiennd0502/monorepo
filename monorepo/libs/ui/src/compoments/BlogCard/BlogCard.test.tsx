import { render } from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { BLOGS } from '@monorepo/constants';

import BlogCard from '.';

const onPress = jest.fn();

const baseProps = {
  item: BLOGS[0],
  onPress
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <BlogCard {...props} />
    </TamaguiProvider>
  );
};

describe('BlogCard', () => {
  it('should render BlogCard component with default props', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render CartItem component with isReview true', () => {
    const container = setup({isReview: true});

    expect(container).toMatchSnapshot();
  });
});
