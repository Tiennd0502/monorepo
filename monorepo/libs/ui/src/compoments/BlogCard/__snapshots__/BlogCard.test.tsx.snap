// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard should render BlogCard component with default props 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "elevation": 16,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 20,
        "marginBottom": 20,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://i.ibb.co/m8nVWx2/sarah-dorweiler-fr0-J5-GIVyg-unsplash-1.png",
        }
      }
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 70,
          "width": 70,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "gap": 6,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          {
            "color": "#606060",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Product 01
      </Text>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": "bold",
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        30.00
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
      }
    }
  >
    <View
      style={
        {
          "rowGap": 5,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 4,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 12,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      20/03/2020
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={5}
    style={
      {
        "color": "#242424",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 14,
        "lineHeight": 20,
        "textAlign": "justify",
      }
    }
    suppressHighlighting={true}
  >
    Nice Furniture with good delivery.Nice Furniture with good delivery. The delivery time is very fast. Then products look like exactly the picture in the app. Besides, color is also the same and quality is very good despite very cheap price
  </Text>
</View>
`;

exports[`BlogCard should render CartItem component with isReview true 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "elevation": 16,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/m8nVWx2/sarah-dorweiler-fr0-J5-GIVyg-unsplash-1.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "height": 40,
        "position": "absolute",
        "top": -20,
        "width": 40,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
      }
    }
  >
    <View
      style={
        {
          "rowGap": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          {
            "color": "#242424",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Product 01
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 4,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="16"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={32}
          vbWidth={32}
          width="16"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0.5,
                    -865972,
                    0.5,
                    -865972,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="a"
                x1="0%"
                x2="100%"
                y1="0%"
                y2="0%"
              />
            </RNSVGDefs>
            <RNSVGPath
              d="M20.388 10.918L32 12.118l-8.735 7.749L25.914 31.4l-9.893-6.088L6.127 31.4l2.695-11.533L0 12.118l11.547-1.2L16.026.6l4.362 10.318z"
              fill={
                {
                  "brushRef": "a",
                  "type": 1,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 12,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      20/03/2020
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={5}
    style={
      {
        "color": "#242424",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 14,
        "lineHeight": 20,
        "textAlign": "justify",
      }
    }
    suppressHighlighting={true}
  >
    Nice Furniture with good delivery.Nice Furniture with good delivery. The delivery time is very fast. Then products look like exactly the picture in the app. Besides, color is also the same and quality is very good despite very cheap price
  </Text>
</View>
`;
